File: app/src/main/java/dev/suhockii/lifetest/ui/main/MainActivity.java
package dev.suhockii.lifetest.ui.main;

import android.os.Bundle;
import android.support.annotation.IdRes;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.ActionBar;
import android.widget.Toast;

import com.arellomobile.mvp.presenter.InjectPresenter;

import com.crashlytics.android.Crashlytics;

import io.fabric.sdk.android.Fabric;

import javax.inject.Inject;

import dev.suhockii.lifetest.R;
import dev.suhockii.lifetest.util.ui.activity.InjectableActivity;
import dev.suhockii.lifetest.util.ui.fragment.InteractionFragment;
import dev.suhockii.lifetest.util.ui.FragmentRouter;

public class MainActivity extends InjectableActivity implements MainView,
        InteractionFragment.OnFragmentInteractionListener {

    @InjectPresenter
    MainPresenter mainPresenter;

    @Inject
    FragmentRouter fragmentRouter;

    private ActionBar actionBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        actionBar = getSupportActionBar();

        if (savedInstanceState == null) {
            if (fragmentRouter == null) {
                Toast.makeText(this, "fragmentRouter null", Toast.LENGTH_SHORT).show();
            } else {
                fragmentRouter.openListFragment();
            }
        }
    }

    @IdRes
    public int getContainerId() {
        return R.id.main_fl_container;
    }

    @Override
    public void onFragmentInteraction(String title) {
        actionBar.setTitle(title);
//        actionBar.setDisplayHomeAsUpEnabled(isDrawerLocked);
    }

    @Override
    public void onBackPressed() {
        FragmentManager fragmentManager = getSupportFragmentManager();

        if (fragmentManager.getBackStackEntryCount() > 1) {
            fragmentManager.popBackStack();
        } else {
            finish();
        }
    }
}

File: app/src/main/java/dev/suhockii/lifetest/ui/main/MainActivity.kt
package dev.suhockii.lifetest.ui.main

import android.os.Bundle
import android.view.MenuItem
import androidx.annotation.IdRes
import dev.suhockii.lifetest.R
import dev.suhockii.lifetest.util.ui.FragmentRouter
import dev.suhockii.lifetest.util.ui.activity.InjectableActivity
import dev.suhockii.lifetest.util.ui.fragment.InteractionFragment
import javax.inject.Inject

class MainActivity : InjectableActivity(), InteractionFragment.OnFragmentInteractionListener {

    @Inject
    lateinit var fragmentRouter: FragmentRouter

    val containerId: Int
        @IdRes
        get() = R.id.main_fl_container

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (savedInstanceState == null) {
            fragmentRouter.openListFragment()
        }
    }

    override fun onFragmentInteraction(title: String, showBackButton: Boolean) {
        supportActionBar!!.title = title
        supportActionBar!!.setDisplayHomeAsUpEnabled(showBackButton)
    }

    override fun onBackPressed() {
        val fragmentManager = supportFragmentManager

        if (fragmentManager.backStackEntryCount > 1) {
            fragmentManager.popBackStack()
        } else {
            finish()
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onBackPressed()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}
